src/Toolkit/Toolkit.UI.Controls/MeasureToolbar/MeasureToolbar.cs

```diff
@@ -354,25 +388,31 @@ private void DisplayResult(Geometry.Geometry? geometry = null)
         }
 
         /// <summary>
-        /// Toggles between measure modes and starts SketchEditor when not already started for length and area.
+        /// Toggles between measure modes and starts <see cref="GeometryEditor"/> when not already started for length and area.
         /// </summary>
         /// <param name="sender">Toggle button that raised click event.</param>
         /// <param name="e">Contains information or event data associated with routed event.</param>
-        private async void OnToggleMeasureMode(object? sender, RoutedEventArgs e)
+        private void OnToggleMeasureMode(object? sender, RoutedEventArgs e)
         {
+            if (_geometryEditor is not null)
+            {
+                _geometryEditor.PropertyChanged -= OnGeometryEditorPropertyChanged;
+                _geometryEditor.PropertyChanged += OnGeometryEditorPropertyChanged;
```

Comment:
Don't ever do -+ thing. Subscribe and unsubscribe in the correct place instead (ie when the editor gets assigned and unassigned)